app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/IvController.scala

```diff
@@ -18,31 +18,44 @@ package uk.gov.hmrc.cgtpropertydisposalsfrontend.controllers
 
 import java.util.UUID
 
+import cats.syntax.eq._
+
 import com.google.inject.{Inject, Singleton}
 import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
 import uk.gov.hmrc.cgtpropertydisposalsfrontend.config.ErrorHandler
 import uk.gov.hmrc.cgtpropertydisposalsfrontend.controllers.actions.{AuthenticatedAction, SessionDataAction, WithAuthAndSessionDataAction}
-import uk.gov.hmrc.cgtpropertydisposalsfrontend.util.Logging
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.SessionData
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.repos.SessionStore
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.util.{Logging, toFuture}
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.util.Logging._
 import uk.gov.hmrc.play.bootstrap.controller.FrontendController
 
+import scala.concurrent.ExecutionContext
+
 @Singleton
 class IvController @Inject()(
+                              sessionStore: SessionStore,
                               val authenticatedAction: AuthenticatedAction,
                               val sessionDataAction: SessionDataAction,
                               errorHandler: ErrorHandler,
                               cc: MessagesControllerComponents
-) extends FrontendController(cc)
-    with WithAuthAndSessionDataAction
-    with Logging {
-
-  def ivSuccess(): Action[AnyContent] = authenticatedActionWithSessionData { implicit request =>
-    request.sessionData.flatMap(_.ivContinueUrl) match {
-      case Some(ivContinueUrl) =>
-        SeeOther(ivContinueUrl)
-
-      case None =>
-        logger.warn("Could not get ivContinueUrl from session")
-        InternalServerError(errorHandler.internalServerErrorTemplate)
+                            )(implicit ec: ExecutionContext) extends FrontendController(cc)
+  with WithAuthAndSessionDataAction
+  with Logging
+  with SessionUpdates {
+
+  def ivSuccess(): Action[AnyContent] = authenticatedActionWithSessionData.async { implicit request =>
```

Comment:
I made it so that we always go to the start endpoint after successful IV. Someone who had to go through the IV journey would not have been going through anywhere other than the start endpoint since there would be no session data to say that they should be somewhere further on in the journey and all of our endpoints should be protected with logic to make sure the session data says they're in the right place.